programs/survival/item_holding_mobs_despawn_rule.sc

```diff
@@ -0,0 +1,21 @@
+// changes despawn rule for monster mobs. 
+// makes mobs who hold items despawnable as opposed to vanilla where they will never despawn.
+// mobs can only be made despawnable when they are given a custom name using nametags
+
+__command()->null;
+
+__config()->{
+    'stay_loaded' -> true;
+};
+
+__change_persistence() -> (
+    for ( entity_list('monster'),
+        if (_ ~ 'persistence' && _ ~ 'holds' && ! _ ~ 'custom_name',
+            modify(_, 'persistence', false);
+        )
+    )
+);
+
+__on_tick()->(
+    if (! tick_time() % 20,  schedule(0, '__change_persistence'));
+);

```

Comment:
I think it would be best to bind this to `entity_load_handler`, rather than a global `__on_tick` call. That wasy you don't need to check any mob more than once (and you also don't use the `__on_tick` method, which is disencouraged).

Reply:
Entity load handler wont work in the case of when a mob picks up item after spawning. For example a zombie picking up block you left behind while mining. or does it ?

Reply:
I think what Firi meant was to attach an `on_tick` entity event to the entity in load handler so there's no need to query the world and every entity every tick, and you are then just checking data in the selected entities.

Reply:
I will look into the documentation.

Reply:
Yes, that's what I meant.` entity_load_handler` also triggers when the entities are laoded in the world, not just newly spawned, iirc, so it should work fine. Don't quote me on this, but hooking up to the `on_tick` event of the mob is less expensive than going through the list of all entities, since the game doesn't need to query the entities. I think, not sure.